be/src/exprs/agg/window.h

```diff
@@ -432,16 +431,33 @@ class LeadLagWindowFunction final : public ValueWindowFunction<PT, LeadLagState<
             return;
         }
 
-        if (columns[0]->is_null(frame_end - 1)) {
-            this->data(state).is_null = true;
-            return;
+        if (!columns[0]->is_null(frame_end - 1)) {
+            this->data(state).is_null = false;
+            const Column* data_column = ColumnHelper::get_data_column(columns[0]);
+            const InputColumnType* column = down_cast<const InputColumnType*>(data_column);
+            AggDataTypeTraits<PT>::assign_value(this->data(state).value,
+                                                AggDataTypeTraits<PT>::get_row_ref(*column, frame_end - 1));
+        } else {
+            if (!ignoreNulls) {
+                this->data(state).is_null = true;
+                return;
+            }
+            // for lead/lag, [peer_group_start, peer_group_end] equals to [partition_start, partition_end]
+            // when lead/lag called, the whole partitoin's data already been here, so we can just check to the begining or the end
+            size_t value_index = useLag ? ColumnHelper::last_nonnull(columns[0], peer_group_start, frame_end - 1)
+                                        : ColumnHelper::find_nonnull(columns[0], frame_end, peer_group_end);
```

Comment:
why `ColumnHelper::find_nonnull(columns[0], frame_end, peer_group_end);` ?

I suppose `frame_end >= peer_group_end` is always positive?

Reply:
for lead/lag, peer_group_end is equal to partition_end, so peer_group_end >= frame_end is always positive. I just use peer_group_end as  partition_end

Reply:
Ok, I misunderstood.